src/main/java/com/evolvedbinary/jnibench/common/array/JniGetArray.java

```diff
@@ -0,0 +1,15 @@
+package com.evolvedbinary.jnibench.common.array;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+public class JniGetArray implements JniListSupplier<FooObject> {
+
+  @Override
+  public List<FooObject> getObjectList(NativeObjectArray<FooObject> nativeObjectArray) {
+    return new ArrayList<>(Arrays.asList(getArray(nativeObjectArray.get_nativeHandle())));
```

Comment:
You don't need the additional `new ArrayList<>` as `Arrays.asList` will already produce a `List<FooObject>`. The superfluous `new ArrayList<>` causes a (potentially expensive) copy of the array to be made

Reply:
I do not know how it got there ;)